music_assistant/providers/ard_audiothek/__init__.py

```diff
@@ -0,0 +1,808 @@
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    # ruff: noqa: ARG001
+    if values is None:
+        values = {}
+
+    authenticated = True
+    if values.get(CONF_TOKEN_BEARER) is None or values.get(CONF_USERID) is None:
+        authenticated = False
+
+    return (
+        ConfigEntry(
+            key="label_text",
+            type=ConfigEntryType.LABEL,
+            label=f"Successfully signed in as {values.get(CONF_USERNAME)} {str(values.get(CONF_EMAIL, '')).replace('@', '(at)')}.",  # noqa: E501
+            hidden=not authenticated,
+        ),
+        ConfigEntry(
+            key=CONF_EMAIL,
+            type=ConfigEntryType.STRING,
+            label="E-Mail",
+            required=False,
+            description="E-Mail address of ARD account.",
+            hidden=authenticated,
+            value=values.get(CONF_EMAIL),
+        ),
+        ConfigEntry(
+            key=CONF_PASSWORD,
+            type=ConfigEntryType.SECURE_STRING,
+            label="Password",
+            required=False,
+            description="Password of ARD account.",
+            hidden=authenticated,
+            value=values.get(CONF_PASSWORD),
+        ),
+        ConfigEntry(
+            key=CONF_MAX_BITRATE,
+            type=ConfigEntryType.INTEGER,
+            label="Maximum bitrate for streams (0 for unlimited)",
+            required=False,
+            description="Maximum bitrate for streams. Use 0 for unlimited",
+            default_value=0,
+            value=values.get(CONF_MAX_BITRATE),
+        ),
+        ConfigEntry(
+            key=CONF_PODCAST_FINISHED,
+            type=ConfigEntryType.INTEGER,
+            label="Percentage reached until podcast episode is marked as finished",
+            required=False,
+            description="This setting defines the percentage of how much of a podcast "
+            "has to be left unheard until an episode is marked as finished",
+            default_value=95,
+            value=values.get(CONF_PODCAST_FINISHED),
+        ),
+        ConfigEntry(
+            key=CONF_TOKEN_BEARER,
+            type=ConfigEntryType.SECURE_STRING,
+            label="token",
+            hidden=True,
+            required=False,
+            value=values.get(CONF_TOKEN_BEARER),
+        ),
+        ConfigEntry(
+            key=CONF_USERID,
+            type=ConfigEntryType.SECURE_STRING,
+            label="uid",
+            hidden=True,
+            required=False,
+            value=values.get(CONF_USERID),
+        ),
+        ConfigEntry(
+            key=CONF_EXPIRY_TIME,
+            type=ConfigEntryType.SECURE_STRING,
+            label="token_expiry",
+            hidden=True,
+            required=False,
+            default_value=0,
+            value=values.get(CONF_EXPIRY_TIME),
+        ),
+        ConfigEntry(
+            key=CONF_USERNAME,
```

Comment:
CONF_USERNAME is typically used for login credentials (the username you type to log in)